common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -355,6 +355,22 @@ public CalendarInterval subtract(CalendarInterval that) {
     return new CalendarInterval(months, microseconds);
   }
 
+  private static CalendarInterval fromDoubles(double months, double microseconds) {
+    long roundedMonths = (long)(months);
+    // Using 30 days per month as PostgreSQL does.
```

Comment:
30 and not 31? just double checking

Reply:
I have double checked in source codes: https://github.com/postgres/postgres/blob/bffe1bd68457e43925c362d8728ce3b25bdf1c94/src/include/datatype/timestamp.h#L77

Reply:
Sounds good.
General question: why are we using PostgreSQL as the point of reference? I'm not objecting to it, just never saw what the motivation was for adding all of the PostgreSQL functionality down to small items like this.

Reply:
The actual reference point for me is the SQL standard which defines 4 arithmetic operations for intervals `+`, `-`, and `*`, `/`. PostgreSQL was taken as **reasonable** reference implementation.

> the motivation was for adding all of the PostgreSQL functionality down to small items like this.

The functionality is small but I would say it is basic. I do believe there are few real use cases for the new operations. For example, if you have 2 timestamp columns, you can calculate difference as an interval column, and calculate average interval, 20% of minimum interval, percentile and so on. Basic arithmetic operations over interval could be useful.

Reply:
Here is operations defined by the SQL standard:
<img width="727" alt="Screen Shot 2019-10-17 at 23 38 37" src="https://user-images.githubusercontent.com/1580697/67045609-54d12a00-f137-11e9-8fdd-c2e9f733f456.png">
By having `*` and `/` for intervals, we cover all interval operations defined by the SQL standard which is not bad, isn't it?

Reply:
Got it, this is a standard operation. Makes sense.